### PR TITLE
Fix ONNX frontend Reshape behavior

### PR DESCRIPTION
**Pull request by trulyspinach**
When performing a 1D to 4D reshape, the current implementation directly perform an in order remap which leads to an unexpected transpose from NCHW(1D) to NHWC thus make the accelerator outputs different result from other ONNX inference engine.

This can be easily reproduced with the following ONNX model:
[reshaper.onnx.zip](https://github.com/tensil-ai/tensil/files/8547055/reshaper.onnx.zip)

Screen Shot 2022-04-23 at 9 48 29 PM

Generated with the following PyTorch code:
```
class Reshaper(nn.Module):
    def __init__(self):
        super(Reshaper, self).__init__()

        self.conv1 = nn.Conv2d(2, 2, 3, stride=1, padding=0)
        self.up = nn.Upsample(scale_factor=2, mode='bilinear', align_corners=False)

    def forward(self, z):
        out = z
        out = z.reshape(z.shape[0], 2,2,2)
        # out = self.up(out)
        return out
```

### CODE REVIEW

**Review by petrohi — CHANGES_REQUESTED** [review]
Looks great! Two suggestions:

- Please format the code according to project's `.scalafmt.conf`.
- (Optional) Please consider adding compiler test that captures this behavior. You'd need to add your test model to https://github.com/tensil-ai/tensil-models. For an example take a look at https://github.com/tensil-ai/tensil/blob/main/tools/test/src/tools/CompilerSpec.scala#L1254.

**Comment by trulyspinach** [comment]
@petrohi Thanks!
I've updated the format and added a test case that checks for the output. 
Could you push the model file here into [tensil-models](https://github.com/tensil-ai/tensil-models)?
[reshape_1d_4d.onnx.zip](https://github.com/tensil-ai/tensil/files/8562381/reshape_1d_4d.onnx.zip)

File: tools/test/src/tools/Reshape.scala [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      .flatten
+      .toArray
+
+    for (i <- 0 until 8)
+      rmse.addSample(result(i), Golden(i))
+
+    assert(rmse.compute < dataType.error)
+  }
+
+  private val Golden = Seq(
```
Comment (petrohi) [inline tools/test/src/tools/Reshape.scala:59]: I suggest asserting on the entire result including zeroes.

**Review by petrohi — CHANGES_REQUESTED** [review]
This is awesome! Please see my suggestion.

**Comment by petrohi** [comment]
I added test model to `tensil-models`.

File: tools/test/src/tools/Reshape.scala (hunk 1 above)
Comment (trulyspinach) [reply tools/test/src/tools/Reshape.scala:64]: Good point! I've fixed it now it checks the full output.

**Review by petrohi — APPROVED** [review]

**Pull request merged**